hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoAdjustAllowanceTransitionLogic.java

```diff
@@ -74,33 +78,31 @@ public CryptoAdjustAllowanceTransitionLogic(
 		this.adjustAllowanceChecks = allowanceChecks;
 		this.dynamicProperties = dynamicProperties;
 		this.sideEffectsTracker = sideEffectsTracker;
+		this.entitiesChanged = new HashMap<>();
 	}
 
 	@Override
 	public void doStateTransition() {
```

Comment:
Didn't get a chance to review earlier. Do we cover the case of an overflow (say maxSupply is set to Long.MAX_VALUE)?

Reply:
@superboo  We check if it exceeds token max supply in Precheck [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/AdjustAllowanceChecks.java#L290) for Adjust and  [here](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/txns/crypto/validators/ApproveAllowanceChecks.java#L253) for approve

Reply:
Looks like overflow will be handled as NEGATIVE_ALLOWANCE_AMOUNT instead of calling it out as overflow. Guess that's ok.